examples/dmd/wave_equation.cpp

```diff
@@ -0,0 +1,630 @@
+   H1_FECollection fe_coll(order, dim);
+   FiniteElementSpace fespace(mesh, &fe_coll);
+
+   //int fe_size = fespace.GetTrueVSize();
+   int true_size = fespace.GetTrueVSize();
+   cout << "Number of temperature unknowns: " << true_size << endl;
+
+    Array<int> true_offset(3);
+    true_offset[0] = 0;
+    true_offset[1] = true_size;
+    true_offset[2] = 2*true_size;
+
+    BlockVector vx(true_offset);
+
+   GridFunction u_gf(&fespace);
+   GridFunction dudt_gf(&fespace);
+   dudt_gf.MakeTRef(&fespace, vx, true_offset[0]);
+   u_gf.MakeTRef(&fespace, vx, true_offset[1]);
+
+
+   // 6. Set the initial conditions for u. All boundaries are considered
+   //    natural.
+   FunctionCoefficient u_0(InitialSolution);
+   u_gf.ProjectCoefficient(u_0);
+   Vector u;
+   u_gf.GetTrueDofs(u);
+
+   FunctionCoefficient dudt_0(InitialRate);
+   dudt_gf.ProjectCoefficient(dudt_0);
+   Vector dudt;
+   dudt_gf.GetTrueDofs(dudt);
+
+   // 7. Initialize the conduction operator and the visualization.
+   Array<int> ess_bdr;
+   if (mesh->bdr_attributes.Size())
+   {
+      ess_bdr.SetSize(mesh->bdr_attributes.Max());
+
+      if (dirichlet)
+      {
+         ess_bdr = 1;
+      }
+      else
+      {
+         ess_bdr = 0;
+      }
+   }
+
+   WaveOperator oper(fespace, ess_bdr, speed);
+
+   u_gf.SetFromTrueDofs(u);
+   {
+      ofstream omesh("wave_equation.mesh");
+      omesh.precision(precision);
+      mesh->Print(omesh);
+      ofstream osol("wave_equation-init.gf");
+      osol.precision(precision);
+      u_gf.Save(osol);
+      dudt_gf.Save(osol);
+   }
+
+   VisItDataCollection visit_dc("Wave_Equation", mesh);
+   visit_dc.RegisterField("solution", &u_gf);
+   visit_dc.RegisterField("rate", &dudt_gf);
+   if (visit)
+   {
+      visit_dc.SetCycle(0);
+      visit_dc.SetTime(0.0);
+      visit_dc.Save();
+   }
+
+   socketstream sout;
+   if (visualization)
+   {
+      char vishost[] = "localhost";
+      int  visport   = 19916;
+      sout.open(vishost, visport);
+      if (!sout)
+      {
+         cout << "Unable to connect to GLVis server at "
+              << vishost << ':' << visport << endl;
+         visualization = false;
+         cout << "GLVis visualization disabled.\n";
+      }
+      else
+      {
+         sout.precision(precision);
+         sout << "solution\n" << *mesh << dudt_gf;
+         sout << flush;
+      }
+   }
+
+   StopWatch fom_timer, dmd_training_timer, dmd_prediction_timer;
+   fom_timer.Start();
+
+   // 8. Perform time-integration (looping over the time iterations, ti, with a
+   //    time-step dt).
+   ode_solver->Init(oper);
+   double t = 0.0;
+
+   vector<double> ts;
+
+   fom_timer.Stop();
+
+   dmd_training_timer.Start();
+   
+   int curr_window = 0;
+   vector<CAROM::DMD*> dmd_u;
+   dmd_u.push_back(new CAROM::DMD(u_gf.GetTrueVector().Size(), dt));
+   dmd_u[curr_window]->takeSample(u_gf.GetTrueVector(), t);
+   ts.push_back(t);
+
+
+   dmd_training_timer.Stop();
+
+   u_gf.SetFromTrueDofs(u);
```

Comment:
This seems unnecessary. Does everything still work if it is removed?

Reply:
Fixed.